### PR TITLE
Xml summary

### PR DESCRIPTION
**Pull request by Withriin**
Added xml summaries and comments.  Also moved battlestatistics dictionary to strategy layer with an separate increment function.

### CODE REVIEW

File: RockPaperScissorsLibrary/RockPaperScissorsFactory.cs [hunk 1]
```diff
@@ -6,20 +6,33 @@
...
             Random computerChoice = new Random();
             int debugComputerChoice = computerChoice.Next(1, weaponDictionary.Count + 1);
             weaponDictionary.TryGetValue(debugComputerChoice.ToString(), out IWeapon computerWeapon);
             return computerWeapon;
         }
 
+        /// <summary>
+        /// Validates user input for player weapon and converts it to an IWeapon.
+        /// </summary>
+        /// <param name="playerWeapon"></param>
```
Comment (Sean-RD-Higgins) [inline RockPaperScissorsLibrary/RockPaperScissorsFactory.cs:34]: This line could have a documentation sentence to help users of the library

File: RockPaperScissorsLibrary/RockPaperScissorsStrategy.cs [hunk 2]
```diff
@@ -7,8 +7,34 @@
 
 namespace RockPaperScissorsLibrary
 {
+    /// <summary>
+    /// Game Rules and statistics
```
Comment (Sean-RD-Higgins) [inline RockPaperScissorsLibrary/RockPaperScissorsStrategy.cs:11]: Perhaps "Handles and defines the"

File: RockPaperScissorsLibrary/RockPaperScissorsStrategy.cs [hunk 3]
```diff
@@ -7,8 +7,34 @@
...
+        /// <returns>Dictionary</returns>
+        public Dictionary<CombatConclusion, int> GetBattleStatisticsMap()
+         {
+             return _battleStatistics;
+         }
+       
+       /// <summary>
+       /// Takes User and computers weapon selections and checks what the outcome was for the User
+       /// </summary>
+       /// <param name="userWeapon"></param>
```
Comment (Sean-RD-Higgins) [inline RockPaperScissorsLibrary/RockPaperScissorsStrategy.cs:35]: Params are yummy

File: RockPaperScissorsLibrary/RockPaperScissorsStrategy.cs [hunk 4]
```diff
@@ -24,5 +50,14 @@ public CombatConclusion GetCombatResult(IWeapon userWeapon, IWeapon computerWeap
                 return CombatConclusion.Defeat;
             }
         }
+        
+        /// <summary>
+        /// Adds one to the players combat conclusion that is inputed
+        /// </summary>
+        /// <param name="combatConclusion"></param>
```
Comment (Sean-RD-Higgins) [inline RockPaperScissorsLibrary/RockPaperScissorsStrategy.cs:57]: yum yum parameters

**Review by Sean-RD-Higgins — CHANGES_REQUESTED** [review]
A couple minor improvements at worst- fantastic

File: RockPaperScissorsLibrary/RockPaperScissorsFactory.cs (hunk 1 above)
Comment (Withriin) [reply RockPaperScissorsLibrary/RockPaperScissorsFactory.cs]: Updated summary.

File: RockPaperScissorsLibrary/RockPaperScissorsStrategy.cs (hunk 2 above)
Comment (Withriin) [reply RockPaperScissorsLibrary/RockPaperScissorsStrategy.cs]: Updated Summary.

File: RockPaperScissorsLibrary/RockPaperScissorsStrategy.cs (hunk 3 above)
Comment (Withriin) [reply RockPaperScissorsLibrary/RockPaperScissorsStrategy.cs]: Updated parameters.

**Review by Sean-RD-Higgins — APPROVED** [review]
Looks good!  Fire away!

**Pull request merged**